src/openhab.cpp

```diff
@@ -173,73 +170,129 @@ void OpenHAB::startSse() {
-            qCDebug(m_logCategory) << "openhab not reachable";
-        } else if (reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt() == 200) {
-            QString answer = reply->readAll();
-            qCDebug(m_logCategory) << reply->header(QNetworkRequest::ContentTypeHeader).toString() <<" : " << answer;
+        _flagOpenHabConnected = false;
+        disconnect();
+        qCDebug(m_logCategory) << "openhab not reachable";
+    } else if (reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt() == 200) {
+        if (state() != CONNECTED) {
+            qCDebug(m_logCategory) << reply->header(QNetworkRequest::ContentTypeHeader).toString() << " : " << answer;
 
             _sseReconnectTimer->setSingleShot(true);
             _sseReconnectTimer->setInterval(2000);
             _sseReconnectTimer->stop();
 
-            QObject::connect(&_sseNetworkManager, &QNetworkAccessManager::finished, this, &OpenHAB::streamFinished);
+            QObject::connect(_sseNetworkManager, &QNetworkAccessManager::finished, this, &OpenHAB::streamFinished);
             QObject::connect(_sseReconnectTimer, &QTimer::timeout, this, &OpenHAB::onSseTimeout);
-
+            _flagOpenHabConnected = true;
             startSse();
-            getItems(true);
+            _firstrun = true;
+            getItems();
+
             setState(CONNECTED);
+
+        } else if (state() == CONNECTED && _flagleaveStandby) {
+            _flagOpenHabConnected = true;
+            _flagleaveStandby = false;
+            getItems();
+            startSse();
+        } else if (state() == CONNECTED && answer.contains("rest/items/")) {
+            QJsonParseError parseerror;
+            if (answer != "") {
+                QJsonDocument doc = QJsonDocument::fromJson(answer.toUtf8(), &parseerror);
+                if (parseerror.error != QJsonParseError::NoError) {
+                    jsonError(parseerror.errorString());
+                    return;
+                }
+                processItems(doc, _firstrun);
+                _firstrun = false;
+
+                _flagOpenHabConnected = true;
+            }
+        } else if (state() == CONNECTED) {
+            _flagOpenHabConnected = true;
         }
-        QObject::disconnect(&_nam, &QNetworkAccessManager::finished, this, 0);
-    });
-    _nam.get(request);
+    } else {
+        m_notifications->add(
+            true, tr("Cannot connect to ").append(friendlyName()).append("."), tr("Reconnect"),
+            [](QObject* param) {
+                Integration* i = qobject_cast<Integration*>(param);
+                i->connect();
+            },
+            this);
+        _flagOpenHabConnected = false;
+        disconnect();
+        qCDebug(m_logCategory) << "openhab not reachable";
+    }
 }
+void OpenHAB::connect() {
+    setState(CONNECTING);
+    do {
+        _tries++;
+        if (_tries == 20) {
+            break;
+        }
+    } while (!manager->isOnline());
```

Comment:
Hello @zehnm I am 70% sure that there is somwhere a bug in the state machine. If you set the state to disconnted and the Entity is "open" the Remote crashes as well as during reconnection if I don't check the network is "up" and "running" the network requests are never recieved. So my suggestin is to check if you can release the reconnection of the integrations after the network is up and running. Second to check during the disconnect state machine if there is any open entity and if so close them. And also to implement a standby state.

Reply:
I think it might be related to a change that was introduced via a PR (can't remember who) that the entity stays open when the remote goes to sleep. It makes sense for use case, but I guess it introduced new issues.

Reply:
Could be if you can point me to the file I can check. @martonborzak